presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowStatsRewrite.java

```diff
@@ -132,13 +133,14 @@ public Visitor(Metadata metadata, Session session, List<Expression> parameters,
         @Override
         protected Node visitShowStats(ShowStats node, Void context)
         {
-            validateShowStats(node);
             checkState(queryExplainer.isPresent(), "Query explainer must be provided for SHOW STATS SELECT");
 
             if (node.getRelation() instanceof TableSubquery) {
                 QuerySpecification specification = (QuerySpecification) ((TableSubquery) node.getRelation()).getQuery().getQueryBody();
+                Plan plan = queryExplainer.get().getLogicalPlan(session, new Query(Optional.empty(), specification, Optional.empty(), Optional.empty()), parameters);
```

Comment:
Why did you pull it up here? You can do this check directly in the `getConstraint`, so you don't have to change `validate` and don't need to extract plan here

Reply:
By pulling it up here,  plan can be reused in both function validateShowStats and getConstraint,  as suggested by @mbasmanova

Reply:
Because of this change,  QuerySpecification is executed before validateShowStats,  which requires a real table 'orders' to parse it successfully.  

Before this diff,  validateShowStats is executed first, thus it is OK to have a non-existing table before hitting an error.